### ISSUE TITLE
Research whether calling `record.data.json()` twice results in a hang

### ISSUE THREAD

**Issue by frankhinek** [TBD54566975/web5-js#202]
@mistermoe had encountered this in the past, and if it still occurs, needs to be fixed.  The cause is that due to the underlying stream being consumed, the second attempt hangs.

**Comment by adam4leos** [comment]
After executing following code, nothing was hanging for me
`
import { Web5 } from '@web5/api';

const { web5, did: myDid } = await Web5.connect();

const { record } = await web5.dwn.records.create({
  data: JSON.stringify({ message: "Hello World!"}),
  message: {
    dataFormat: "application/json",
  },
});

const { records } = await web5.dwn.records.query({
  message: {
    filter: {
      dataFormat: "application/json",
    },
  },
});

const {data} = records[0];

const res1 = await data.json();
console.log('111111-----', res1);

const res2 = await data.json();
console.log('222222------', res2)

const res3 = await data.json();
console.log('333333-----', res3);

const res4 = await data.json();
console.log('444444------', res4)`

**Comment by adam4leos** [comment]
While there was no hanging issue, note that output of the data after `.json()` is different in latest `main` branch. Looks like a problem to me, unless it's part of the plan @frankhinek  @mistermoe 

Let me know if that something WIP or you'd like me to investigate. 

Quick look up show that `dataBlob` differ in branches:

```typescript
{ dataBlob: Blob { size: 8490, type: 'application/json' } } // latest main
{ dataBlob: Blob { size: 28, type: 'application/json' } } // v0.7.11
```

**Comment by adam4leos** [comment]
> While there was no hanging issue, note that output of the data after `.json()` is different in latest `main` branch. Looks like a problem to me, unless it's part of the plan @frankhinek @mistermoe
> 
> Let me know if that something WIP or you'd like me to investigate.
> 
> Quick look up show that `dataBlob` differ in branches:
> 
> ```ts
> { dataBlob: Blob { size: 8490, type: 'application/json' } } // latest main
> { dataBlob: Blob { size: 28, type: 'application/json' } } // v0.7.11
> ```

this was clarified in the [discord thread ](https://discord.com/channels/937858703112155166/969272658501976117/1149114464688427119)... 

TLDR - now there are more records returned through generic query, namely identity records, so confusion was that created data record wasn't the first `[0]` item in the array of records...
